### PR TITLE
prepare pom for public to maven repository

### PR DESCRIPTION
**Pull request by MarinaRazumovsky**

### CODE REVIEW

File: sawmill-core/pom.xml [hunk 1]
```diff
@@ -10,7 +10,7 @@
     <parent>
         <artifactId>sawmill</artifactId>
         <groupId>io.logz.sawmill</groupId>
-        <version>1.1.42</version>
+        <version>1.1.43</version>
```
Comment (asafm) [inline sawmill-core/pom.xml]: @MarinaRazumovsky @oririechman I've noticed that a popular pattern is being used for versioning libraries:
1. They use a 0-SNAPSHOT version in all pom.xml files
2. In travis they define a deploy stage that is activated only when a new GitHub release is created. It runs a a script which changes the maven version to the GitHub release name (e.g. 1.2.4) and then uploads it to the repository. 
See: 
* https://github.com/testcontainers/testcontainers-java/blob/master/.travis.yml
* https://maven.apache.org/maven-ci-friendly.html

Thus we can also utilise the Release tab of GitHub for keeping track of release notes.

File: CHNGELOG.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Change Log
```
Comment (asafm) [inline CHNGELOG.md]: I thought we decided to use the Releases by GitHub?

File: RELEASING.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Release process
+
+Sawmill' release process is automated as a Travis deployment. This describes the basic steps for a project member to perform a release.
```
Comment (asafm) [inline RELEASING.md]: project admin, not member

File: RELEASING.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+## Steps
+
+1. Ensure that the master branch is building and that tests are passing.
+1. Ensure that the [`CHANGELOG`](CHANGELOG.md) file is up to date and includes all merged features.
```
Comment (asafm) [inline RELEASING.md]: As I wrote above - maybe use only GitHub releases?

File: RELEASING.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+1. Release of published artifacts is fully automated. Once the Travis build completes, there are no further actions to perform on the repository.
+
+## Internal details
+
+* The signing and publishing steps are initiated as a script-type Travis CI deployment phase.
+* mvn -Drevision=$TRAVIS_TAG is applied to use the tag name to the version number for the build.
+* An encrypted GPG key within the release directory is used for signing. This key file is decrypted using Travis secrets.
+* Travis secrets also hold GPG and Maven Central API username/passwords that are used for publishing.
+* The publication process is automated using nexus-staging-maven-plugin.
+* The process is done with adle and Bintray.
```
Comment (asafm) [inline RELEASING.md]: When you copy paste, you have to make sure you read every line :)

File: RELEASING.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+* Travis service will automatically promote the release to Maven Central.
```
Comment (asafm) [inline RELEASING.md]: Let's unify this with "The publication process is automated using nexus-staging-maven-plugin"
Maybe
* The publish/deploy to Maven Central is done via nexus-staging-maven-plugin

File: RELEASING.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+1. Ensure that the master branch is building and that tests are passing.
+1. Ensure that the [`CHANGELOG`](CHANGELOG.md) file is up to date and includes all merged features.
+1. Create a new release on GitHub. **The tag name is used as the version**, so please keep the tag name plain (e.g. 1.2.3).
+1. Check that the Travis build passed.
+1. Release of published artifacts is fully automated. Once the Travis build completes, there are no further actions to perform on the repository.
+
+## Internal details
+
+* The signing and publishing steps are initiated as a script-type Travis CI deployment phase.
```
Comment (asafm) [inline RELEASING.md]: * The signing the publishing to Maven Central steps are initiated as a script-type Travis CI "deploy" phase, upon a new git tag. The tag is created when a release is created via GitHub.

File: pom.xml [hunk 8]
```diff
@@ -60,6 +60,7 @@
         <assertj.version>3.5.2</assertj.version>
         <logback.version>1.1.7</logback.version>
         <commonsIO.version>2.5</commonsIO.version>
+        <revision>1.1.43-SNAPSHOT</revision>
```
Comment (asafm) [inline pom.xml]: Why an explicit number - it shows as if we're going to advance this number. Maybe something like 0-SNAPSHOT is better ?
Also, let's include a simple explanation:
```
<!--
   The version of this project is determined upon deployment using a a GitHub release name (i.e. tag). The version mentioned here is not a real version, and serves mainly for local builds
-->
```

File: sawmill-benchmark/pom.xml [hunk 9]
```diff
@@ -42,7 +42,7 @@
         <dependency>
             <groupId>io.logz.sawmill</groupId>
             <artifactId>sawmill-core</artifactId>
-            <version>${project.version}</version>
+            <version>${revision}</version>
```
Comment (asafm) [inline sawmill-benchmark/pom.xml]: This reminded me - do we have a plan to automate the benchmark build?

File: CHNGELOG.md (hunk 2 above)
Comment (MarinaRazumovsky) [reply CHNGELOG.md]: removed

File: RELEASING.md (hunk 3 above)
Comment (MarinaRazumovsky) [reply RELEASING.md]: ok

File: RELEASING.md (hunk 4 above)
Comment (MarinaRazumovsky) [reply RELEASING.md]: removed

File: RELEASING.md (hunk 6 above)
Comment (MarinaRazumovsky) [reply RELEASING.md]: ok

File: pom.xml (hunk 8 above)
Comment (MarinaRazumovsky) [reply pom.xml]: ok

File: sawmill-benchmark/pom.xml (hunk 9 above)
Comment (MarinaRazumovsky) [reply sawmill-benchmark/pom.xml]: no, but it needs version

**Pull request merged**